src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -0,0 +1,680 @@
+        ReturnErrorOnFailure(reader.Get(mStorageId.mGroupId));
+        ReturnErrorOnFailure(reader.Next(TLV::ContextTag(kTagSceneID)));
+        ReturnErrorOnFailure(reader.Get(mStorageId.mSceneId));
+
+        // Scene Data
+        ReturnErrorOnFailure(reader.Next());
+        TLV::Tag currTag = reader.GetTag();
+        VerifyOrReturnError(TLV::ContextTag(kTagSceneName) == currTag || TLV::ContextTag(kTagSceneDTransitionTime) == currTag,
+                            CHIP_ERROR_WRONG_TLV_TYPE);
+
+        // If there was no error, a name is expected from the storage, if there was an unexpectec TLV element,
+        if (currTag == TLV::ContextTag(kTagSceneName))
+        {
+            ReturnErrorOnFailure(reader.Get(NameSpan));
+            mStorageData.SetName(NameSpan);
+            ReturnErrorOnFailure(reader.Next(TLV::ContextTag(kTagSceneDTransitionTime)));
+        }
+
+        ReturnErrorOnFailure(reader.Get(mStorageData.mSceneTransitionTimeSeconds));
+        ReturnErrorOnFailure(reader.Next(TLV::ContextTag(kTagSceneDTransitionTime100)));
+        ReturnErrorOnFailure(reader.Get(mStorageData.mTransitionTime100ms));
+        ReturnErrorOnFailure(mStorageData.mExtensionFieldSets.Deserialize(reader));
+
+        return reader.ExitContainer(container);
+    }
+};
+
+// A Full fabric serialized TLV length is 88 bytes, 128 bytes gives some slack tested bu running writer.GetLengthWritten at the
+// end of the Serialize method of FabricSceneData
+static constexpr size_t kPersistentFabricBufferMax = 128;
+
+/**
+ * @brief Linked list of all scenes in a fabric, stored in persistent memory
+ *
+ * FabricSceneData is an access to a linked list of scenes
+ */
+struct FabricSceneData : public PersistentData<kPersistentFabricBufferMax>
+{
+    FabricIndex fabric_index = kUndefinedFabricIndex;
+    uint8_t scene_count      = 0;
+    SceneStorageId scene_map[kMaxScenePerFabric];
+    FabricIndex next = kUndefinedFabricIndex;
+
+    FabricSceneData() = default;
+    FabricSceneData(FabricIndex fabric) : fabric_index(fabric) {}
+
+    CHIP_ERROR UpdateKey(StorageKeyName & key) override
+    {
+        VerifyOrReturnError(kUndefinedFabricIndex != fabric_index, CHIP_ERROR_INVALID_FABRIC_INDEX);
+        key = DefaultStorageKeyAllocator::FabricScenesKey(fabric_index);
+        return CHIP_NO_ERROR;
+    }
+
+    void Clear() override
+    {
+        scene_count = 0;
+        for (uint8_t i = 0; i < kMaxScenePerFabric; i++)
+        {
+            scene_map[i].Clear();
+        }
+        next = kUndefinedFabricIndex;
+    }
+
+    CHIP_ERROR Serialize(TLV::TLVWriter & writer) const override
+    {
+        TLV::TLVType container;
+        ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, container));
+
+        ReturnErrorOnFailure(writer.Put(TLV::ContextTag(kTagSceneCount), static_cast<uint16_t>(scene_count)));
+
+        // Storing the scene map
+        for (uint8_t i = 0; i < kMaxScenePerFabric; i++)
```

Comment:
But not all of those are initialized, right?  Do we still want to store/read all of them?